HW4_Gorbarenko/amino_analyzer.py

```diff
@@ -0,0 +1,196 @@
+        monoisotopic_weights = [71.03711, 156.10111, 114.04293, 115.02694, 103.00919, 129.04259, 128.05858, 57.02146, 137.05891, 113.08406,
+                                113.08406, 128.09496, 131.04049, 147.06841, 97.05276, 87.03203, 101.04768, 186.07931, 163.06333, 99.06841]
+        weights_aa = monoisotopic_weights
+    else:
+        raise ValueError(f"I do not know what '{weight}' is :( \n Read help or just do not write anything except your sequence")
+
+    return weights_aa
+
+
+def aa_weight(seq: str, weight: str = 'average') -> float:
+    """
+    Calculate the amino acids weight in a protein sequence.
+
+    Args:
+        seq (str): The amino acid sequence to calculate the weight for.
+        weight (str, optional): The type of weight to use, either 'average' or 'monoisotopic'. Default is 'average'.
+
+    Returns:
+        float: The calculated weight of the amino acid sequence.
+    """
+    aa_list = str('A, R, N, D, C, E, Q, G, H, I, L, K, M, F, P, S, T, W, Y, V').split(', ')
+    weights_aa = choose_weight(weight)
+    aa_to_weight = dict(zip(aa_list, weights_aa))
+    final_weight = 0
+    for i in seq.upper():
+        final_weight += aa_to_weight.get(i, 0)
+    return round(final_weight, 3)
+
+
+def count_hydroaffinity(seq: str) -> tuple:
+    """
+    Count the quantity of hydrophobic and hydrophilic amino acids in a protein sequence.
+
+    Args:
+        seq (str): The protein sequence for which to count hydrophobic and hydrophilic amino acids.
+
+    Returns:
+        tuple: A tuple containing the count of hydrophobic and hydrophilic amino acids, respectively.
+    """
+    hydrophobic_aa = ['A', 'V', 'L', 'I', 'P', 'F', 'W', 'M']
+    hydrophilic_aa = ['R', 'N', 'D', 'C', 'Q', 'E', 'G', 'H', 'K', 'S', 'T', 'Y']
+    
+    hydrophobic_count = 0
+    hydrophilic_count = 0
+    
+    seq = seq.upper()
+    
+    for aa in seq:
+        if aa in hydrophobic_aa:
+            hydrophobic_count += 1
+        elif aa in hydrophilic_aa:
+            hydrophilic_count += 1
+    
+    return hydrophobic_count, hydrophilic_count
+
+def peptide_cutter(sequence: str, enzyme: str = "trypsin") -> str:
+    """
+    This function identifies cleavage sites in a given peptide sequence using a specified enzyme.
+    
+    Args:
+        sequence (str): The input peptide sequence.
+        enzyme (str): The enzyme to be used for cleavage. Choose between "trypsin" and "chymotrypsin". Default is "trypsin".
+        
+    Returns:
+        str: A message indicating the number and positions of cleavage sites, or an error message if an invalid enzyme is provided.
+    """
+    cleavage_sites = []
+    if enzyme not in ("trypsin", "chymotrypsin"):
+        return "You have chosen an enzyme that is not provided. Please choose between trypsin and chymotrypsin."
+    
+    if enzyme == "trypsin":  # Trypsin cuts peptide chains mainly at the carboxyl side of the amino acids lysine or arginine.
+        for i in range(len(sequence)-1):
```

Comment:
Уделите больше внимания неймингу. i - плохой нейминг здесь.
```suggestion
        for i in range(len(sequence) - 1):
```